src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColService.java

```diff
@@ -75,7 +78,8 @@ public Optional<String> addFullSequenceCollection(
         if (numSeqCols > 0) {
             logger.warn("SeqCol with digest " + levelOneEntity.getDigest() + " already exists !");
             throw new DuplicateSeqColException(levelOneEntity.getDigest());
-        } else {
+        }else {
+            levelOneEntity.getMetadata().forEach(md -> md.setSeqColLevelOne(levelOneEntity));
```

Comment:
@nitin-ebi @apriltuesday 
Just wanna confirm that this is right implementation, because it didn't run otherwise, it returned an error:
```bash
org.hibernate.PropertyValueException: not-null property references a null or transient value : uk.ac.ebi.eva.evaseqcol.entities.SeqColMetadataEntity.seqColLevelOne
```
Originally, I thought that the initialization would be made by `JPA` automatically.

Reply:
Sorry we missed this question when we met yesterday... Others might be more familiar with Hibernate/JPA, but I think this seems right - it's similar to [this question](https://stackoverflow.com/questions/6389600/not-null-property-references-a-null-or-transient-value) where the answers suggest you do have to set the parent explicitly.